saos-webapp/src/test/java/pl/edu/icm/saos/webapp/analysis/csv/ChartCsvExporterTest.java

```diff
@@ -0,0 +1,87 @@
+package pl.edu.icm.saos.webapp.analysis.csv;
+
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+import java.io.IOException;
+import java.io.Writer;
+
+import org.assertj.core.util.Lists;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+
+import pl.edu.icm.saos.common.chart.Chart;
+import pl.edu.icm.saos.common.chart.Series;
+import au.com.bytecode.opencsv.CSVWriter;
+
+/**
+ * @author Łukasz Dumiszewski
+ */
+
+public class ChartCsvExporterTest {
+
+    @InjectMocks
+    private ChartCsvExporter chartCsvExporter = spy(new ChartCsvExporter());
+    
+    @Mock
+    private ChartCsvGenerator chartCsvGenerator;
+    
+    @Mock
+    private CSVWriter csvWriter;
+    
+    
+    @Before
+    public void before() {
+        
+        initMocks(this);
+    }
+    
+    
+    //------------------------ TESTS --------------------------
+    
+    @Test
+    @SuppressWarnings("unchecked")
+    public void exportChartToCsv() throws IOException {
+        
+        // given
+        
+        Writer writer = mock(Writer.class);
+        doReturn(csvWriter).when(chartCsvExporter).createCsvWriter(writer); 
```

Comment:
In such a case I wouldn't be able to check these:
       verify(csvWriter).writeNext(header);
       verify(csvWriter).writeNext(row1);
       verify(csvWriter).writeNext(row2);
       verify(csvWriter).close(); <-- especially this

I wanted to focus on checking if CSVWriter methods are called rather than if Writer gets what it should, but of course your solution also has some advantages.